src/core/utils/fs.cpp

```diff
@@ -112,57 +113,53 @@ QString Utils::Fs::folderName(const QString& file_path)
 }
 
 /**
- * Remove an empty folder tree.
- *
- * This function will also remove .DS_Store files on Mac OS and
- * Thumbs.db on Windows.
+ * This function will first remove system cache files, e.g. `Thumbs.db`,
+ * `.DS_Store`. Then will try to remove the whole tree if the tree consist
+ * only of folders
  */
-bool Utils::Fs::smartRemoveEmptyFolderTree(const QString& dir_path)
+bool Utils::Fs::smartRemoveEmptyFolderTree(const QString& path)
 {
-    qDebug() << Q_FUNC_INFO << dir_path;
-    if (dir_path.isEmpty())
+    if (!QDir(path).exists())
         return false;
 
-    QDir dir(dir_path);
-    if (!dir.exists())
-        return true;
-
-    // Remove Files created by the OS
-#if defined Q_OS_MAC
-    forceRemove(dir_path + QLatin1String("/.DS_Store"));
-#elif defined Q_OS_WIN
-    forceRemove(dir_path + QLatin1String("/Thumbs.db"));
+    static const QStringList deleteList = {
+#if defined Q_OS_WIN
+        "Thumbs.db", "desktop.ini"
+#elif defined Q_OS_LINUX
+        ".directory"
+#elif defined Q_OS_MAC
+        ".DS_Store"
```

Comment:
Quick review. I'll try tomorrow for full review.
These files should be removed on every platform. (eg the user could have booted into Linux and mounted the partition, which creates ".directory".